app/frontend/src/features/donations/DonationsBoxMixed.tsx

```diff
@@ -0,0 +1,458 @@
+      const stripe = (await stripePromise)!;
+      const session_id = await service.donations.initiateDonation(formData);
+      // When the customer clicks on the button, redirect them to Checkout.
+      const result = await stripe.redirectToCheckout({
+        sessionId: session_id,
+      });
+      if (result.error) {
+        throw Error(result.error.message);
+      }
+    },
+    {
+      onSuccess: () => {
+        resetForm();
+      },
+    }
+  );
+
+  const onSubmit = handleSubmit((data) => {
+    initiateDonation(data);
+  });
+
+  return (
+    <>
+      <form onSubmit={onSubmit} className={classes.donationsBox}>
+        {error && <Alert severity="error">{error.message}</Alert>}
+        {success && (
+          <Alert severity="success">
+            Thank you for donating to Couchers.org! We appreciate your support!
+          </Alert>
+        )}
+        {cancelled && (
+          <Alert severity="warning">The payment was cancelled.</Alert>
+        )}
+        <Typography className={classes.marginBottom2} variant="h3">
+          {DONATIONSBOX_TITLE}
+        </Typography>
+        <Controller
+          id="recurring"
+          control={control}
+          name="recurring"
+          rules={{ required: "This field is required." }}
+          defaultValue="monthly"
+          render={({ onChange, value }) => (
+            <FormControl className={classes.formGroup}>
+              <RadioGroup
+                className={classes.donationsBoxRow}
+                aria-label="recurring"
+                name="recurring-radio"
+                onChange={(e) => onChange(e.target.value === "monthly")}
+                value={value ? "monthly" : "one-time"}
+              >
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="monthly"
+                  control={<Radio />}
+                  label={DONATIONSBOX_MONTHLY}
+                />
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="one-time"
+                  control={<Radio />}
+                  label={DONATIONSBOX_ONETIME}
+                />
+              </RadioGroup>
+              <FormHelperText error={!!errors?.recurring?.message}>
+                {errors?.recurring?.message ??
+                  "A monthly gift helps us plan ahead and build our operations on a stable footing."}
+              </FormHelperText>
+            </FormControl>
+          )}
+        />
+
+        <Divider className={classes.marginY2} />
+
+        <div className={classes.donationsBoxRow}>
+          <div className={classes.donationsBoxSubRow}>
+            <button
+              className={`${classes.buttonSecondary} ${
+                singleAmount === DONATIONSBOX_VALUES[0].amount
+                  ? "buttonSecondaryActive"
+                  : ""
+              }`}
+              value={DONATIONSBOX_VALUES[0].amount}
```

Comment:
These buttons currently won't set the value in `react-hook-form` because they aren't input components (value doesn't do anything on buttons).
You could wrap them all, and the custom input, in a `<Control render={({onChange, value}) => ...}>` and then call `onChange(amount)` when you want to change the amount (ie. the buttons `onClick` or the custom input's `onChange`), and then you can also have access to the current value from `value` without needing `watch`

Reply:
Here you mean `Controller`, isn't it?

Reply:
I'm a little bit lost on how to do the onclick event, I made a push so you can take a look.

Reply:
Yes I mean controller. I will take a look tonight, but it's something like this:
```
<Controller control={control} name="amount" render={({ onChange, value }) => (
  <>
    <MyButton onClick={() => onChange(5)} isActive={value === 5} ...>
    <MyButton onClick={() => onChange(10)} isActive={value === 10} ...>
  </>
)} />
```